plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.test.tsx

```diff
@@ -67,14 +94,68 @@ const sampleEntities: Entity[] = [
   },
 ];
 
+const getEntitiesByRefs = jest.fn(async ({ entityRefs }) => ({
+  items: entityRefs.map((e: string) =>
+    ownerEntities.find(f => f.metadata.name === e),
+  ),
+}));
+const mockCatalogApi: Partial<CatalogApi> = {
+  getEntitiesByRefs,
+};
+
 describe('<EntityOwnerPicker/>', () => {
-  it('renders all owners', () => {
-    render(
-      <MockEntityListContextProvider
-        value={{ entities: sampleEntities, backendEntities: sampleEntities }}
-      >
-        <EntityOwnerPicker />
-      </MockEntityListContextProvider>,
+  const mockApis = TestApiRegistry.from([catalogApiRef, mockCatalogApi]);
+
+  it('renders display name when available', async () => {
+    const names: Record<string, string> = {
+      'some-owner': 'Some Team',
+      'some-owner-2': 'Other Team',
+      'another-owner': 'AnotherTeam',
+    };
+    getEntitiesByRefs.mockResolvedValueOnce({
+      items: ownerEntities.map(e => {
+        e.spec = {
+          ...e.spec,
+          profile: {
+            displayName: names[e.metadata.name],
```

Comment:
This feels awkward; it's the exact same value as the name already - why doesn't your example data at the top just contain some entities that have a display name, some that have a title, and some that don't have either of those (all different)? Then you can just check that they all get rendered in the expected way.

Reply:
Yup, I was trying to keep the existing test cases as is, but it makes sense to just shift this to the values returned from the API.